server/handlers/shared_validators.js

```diff
@@ -10,6 +10,19 @@ exports.checkValidationErrorMiddleware = (req, res, next) => {
   next();
 };
 
+exports.clientIDPathValidator = body("clientId").isMongoId();
+exports.productsValidator = body("products").isArray();
```

Comment:
products is not an array from design. you didn't check that the products object contains at least one item and maximum (a reasonable number here)